### PR TITLE
Upgrade Hudi version

### PR DESCRIPTION
**Pull request by codope**
Upgrade the version of `hudi-presto-bundle` from 0.12.1 to 0.13.1.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/pom.xml
Comment (pratyakshsharma) [inline presto-spark-base/pom.xml:310]: Why are we adding these classes as ignored?

**Review by pratyakshsharma — COMMENTED** [review]
Hi @codope , just trying to understand if bumping the version fixes something, or we are doing this as a regular exercise. 
Also it will be good to verify MoR queries once manually.

**Comment by codope** [comment]
@pratyakshsharma We need to include this fix for MOR (hudi connector makes use of the custom record reader for MOR) - https://github.com/apache/hudi/commit/cd314b8cfa58c32f731f7da2aa6377a09df4c6f9

File: presto-spark-base/pom.xml
Comment (pratyakshsharma) [reply presto-spark-base/pom.xml:310]: Yeah, that's right. I tried building on my local. This is needed to fix the build.

**Comment by agrawalreetika** [comment]
Hi @codope & @pratyakshsharma , 
Could we also check this issue https://github.com/prestodb/presto/issues/19166 which is caused by `Caused by: java.lang.ClassNotFoundException: com.esotericsoftware.reflectasm.ConstructorAccess` and upgrade hudi budle to version which also include Jar including this missing class?

**Comment by pratyakshsharma** [comment]
@agrawalreetika Upgrading the version to 0.13.0 will solve #19166. I have verified this.

**Comment by pratyakshsharma** [comment]
@codope Once you confirm MoR and CoW queries are fine, this is good to merge. Thank you for upgrading the version.

**Comment by codope** [comment]
@pratyakshsharma Yes, ive tested for COW and MOR. It works fine with the upgrade. The integration test is failing due to build issue. The builds runs succesfully for me locally. I will check what's going wrong.
```
Warning:  Found duplicate and different resources in [org.apache.httpcomponents:httpclient:4.5.5, org.apache.hudi:hudi-presto-bundle:0.13.0]:
Warning:    mozilla/public-suffix-list.txt
Warning:  Found duplicate classes/resources in compile classpath.
Warning:  Found duplicate and different resources in [org.apache.httpcomponents:httpclient:4.5.5, org.apache.hudi:hudi-presto-bundle:0.13.0]:
Warning:    mozilla/public-suffix-list.txt
Warning:  Found duplicate classes/resources in runtime classpath.
Warning:  Found duplicate and different resources in [org.apache.httpcomponents:httpclient:4.5.5, org.apache.hudi:hudi-presto-bundle:0.13.0]:
Warning:    mozilla/public-suffix-list.txt
Warning:  Found duplicate classes/resources in test classpath.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  19.214 s
[INFO] Finished at: 2023-03-08T16:24:43Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.basepom.maven:duplicate-finder-maven-plugin:1.2.1:check (default) on project presto-hive-hadoop2: Found duplicate classes/resources! -> [Help 1]
```

**Comment by codope** [comment]
precisely that's what i was waiting for. i already have it in my fork (tested with rc1). will update this PR once the artifacts are released.

**Comment by codope** [comment]
@pratyakshsharma @agrawalreetika Can you help me with the [linux-spark-e2e-tests](https://app.circleci.com/pipelines/github/prestodb/presto/4460/workflows/b631a153-0a54-48a7-8651-a2f0027ce3b0/jobs/9905) failure? I am not sure how this patch can cause failure in unrelated test. The exclusion of `org.objenesis` module is only restricted to `hudi-presto-bundle` which is not used in this test I believe.

**Comment by pratyakshsharma** [comment]
@codope ack. Even I am facing issues with these e2e tests. However in your case, the logs are not generated properly. If you can trigger the tests again, it will be helpful.

**Comment by pratyakshsharma** [comment]
Please rebase on latest master, one fix was merged yesterday for a flaky test - https://github.com/prestodb/presto/pull/20049/files#diff-d90218dfbe8eceb0cb6a73620de67e9a15565fb7ab364c09ea59fa1a29d7b39c @codope

**Comment by pratyakshsharma** [comment]
For linux-presto-e2e-tests - https://github.com/prestodb/presto/pull/20228

**Comment by codope** [comment]
@agrawalreetika @pratyakshsharma Can we land this? I don't think the test failures are related.

**Comment by imjalpreet** [comment]
@codope I have retriggered the failing tests as they are required checks before any PR can be merged.

File: presto-native-execution/pom.xml
Comment (imjalpreet) [inline presto-native-execution/pom.xml:98]: Any reason why we are excluding these from this version? Since they were not excluded from any older Hudi version? The linux-presto-e2e-tests failures are probably related.

Comment (codope) [reply presto-native-execution/pom.xml:98]: oh i excluded that because `TestPrestoSparkNativeJoinQueries` in presto-native-execution was failing due to `java.lang.NoClassDefFoundError: org/objenesis/strategy/InstantiotorStrategy`. After comparing the dependency tree of presto-native-execution for this patch vs master, the only additional thing was `kryo-shaded` coming from hudi-presto-bundle. I thought tests in presto-native execution won't need hudi-presto-bundle so I excluded it. Let me replace this exclusion by 
```
<exclusion>
    <groupId>com.esotericsoftware</groupId>
    <artifactId>kryo-shaded</artifactId>
</exclusion>
```
instead of whole bundle.

**Comment by agrawalreetika** [comment]
> @agrawalreetika @pratyakshsharma Can we land this? I don't think the test failures are related. I am trying to run `TestPrestoSparkNativeJoinQueries` but test setup fails in init stage.

@codope , I see tests are failing via `presto-native-execution` module with below error  :
`java.lang.NoClassDefFoundError: org/apache/hudi/hadoop/HoodieParquetInputFormat`
I think we should verify the exclusions made in `presto-native-execution`  pom, and make sure HoodieParquetInputFormat class is getting included?

**Comment by codope** [comment]
Fixed the issue and all checks are successful. @imjalpreet @agrawalreetika Can we merge this PR?

**Review by pratyakshsharma — APPROVED** [review]
Please squash the commits @codope

**Comment by pratyakshsharma** [comment]
Looks like we need to retrigger the build. 

@prestodb/committers This should be good for final pass.

**Review by tdcmeehan — APPROVED** [review]

**Comment by tdcmeehan** [comment]
#20515 will make `test-other-modules` less flaky.  Appreciate the patience, and any help on fixing flaky tests.

**Comment by imjalpreet** [comment]
@codope Can you please do a final rebase? The PR mentioned above to help reduce flakiness/timeout of `test-other-modules` has been merged. Thanks

**Comment by codope** [comment]
@imjalpreet Rebased. Can you please re-trigger `product-tests-specific-environment2`?

**Review by agrawalreetika — APPROVED** [review]

**Review by imjalpreet — APPROVED** [review]

**Comment by codope** [comment]
@tdcmeehan @imjalpreet @agrawalreetika @pratyakshsharma Thanks for the review. Please help in getting patch landed.

**Comment by rschlussel** [comment]
> @shrinidhijoshi @shangxinli could you please take a review? We need code owner approval to merge it :)

I think this is my bad. Should have added prestodb/committers as codeowners for presto-on-spark Fixing it here https://github.com/prestodb/presto/pull/20540

**Pull request merged**
